app/src/main/kotlin/com/adriangl/pokeapi_mvvm/pokemonlist/PokemonListViewModel.kt

```diff
@@ -37,17 +38,28 @@ class PokemonListViewModel(app: Application) : RxAndroidViewModel(app), KodeinAw
         }
     }
 
-    fun getPokemonListLiveData() = pokemonListLiveData
+    fun getPokemonListLiveData(): LiveData<PokemonListViewData> = pokemonListLiveData
+
+    @VisibleForTesting
+    fun getPokemonMutableLiveData() = pokemonListLiveData
 
     fun getPokemonDetails() {
-        dispatcher.dispatch(GetPokemonDetailsListAction())
+        if (pokemonListLiveData.value == null ||
```

Comment:
I'd keep the null checks out of the method in case we want to unconditionally retry the action if called from the Activity.

Also, in case we want to do this (for example, when initializing the ViewModel), I think it would be a better idea to do this checking the store's states rather than with the LiveData so we can simplify the checks to "do we have data in the store to convert it to our resource?". Also, I think we shouldn't rely on checking the LiveData's value because that would make the ViewModel rely on two sources of state: the Store and the LiveData; I'd rather keep the LiveData as a way to pass data from the Store to the view.